src/stores/SymbolListContext.tsx

```diff
@@ -0,0 +1,48 @@
+import { createContext, useContext } from 'react';
+import { useQuery } from '@tanstack/react-query';
+import { SymbolList, fetchSymbolList } from './fetchSymbolList';
+
+type SymbolListContextProps = {
+  symbolList: SymbolList[] | undefined;
+  isLoading: boolean;
+  isError: boolean;
+};
+
+const SymbolListContext = createContext<SymbolListContextProps | undefined>(
+  undefined,
+);
+
+export const SymbolListProvider = ({
+  children,
+}: {
+  children: React.ReactNode;
+}) => {
+  const {
+    data: symbolList,
+    isLoading,
+    isError,
+  } = useQuery({
+    queryKey: ['symbolList'],
+    queryFn: fetchSymbolList,
+    staleTime: Infinity,
+    gcTime: Infinity,
+  });
```

Comment:
This provider could be just a hook called `useSymbolList`. Then in here you would want to check for DB data and decide if the given query should be enabled. `useQuery` has a parameter `enabled` that you can use for that